Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageGiphyView.swift

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Nuke
+import StreamChat
+import SwiftyGif
+import UIKit
+
+open class ChatMessageGiphyView<ExtraData: ExtraDataTypes>: View, UIConfigProvider {
+    public var content: _ChatMessageAttachment<ExtraData>? {
+        didSet {
+            guard oldValue?.imageURL != content?.imageURL else { return }
```

Comment:
What if old and new imageURLs match but previous downloading has failed? 🤔
As an option we can check if image view is displaying gif and if not - download even though urls are the same.

Reply:
d1e2405